client/landing/stepper/hooks/use-flow-login-url.ts

```diff
@@ -0,0 +1,43 @@
+import { addQueryArgs } from '@wordpress/url';
+import { useLoginUrl } from '../utils/path';
+import { useFlowLocale } from './use-flow-locale';
+import type { Flow } from 'calypso/landing/stepper/declarative-flow/internals/types';
+
+type UseFlowLoginUrlProps = {
+	flow: Flow;
+	loginUrlParams?: Record< string, string | number >;
+	pageTitle?: string;
+	returnStepSlug?: string;
+	returnUrlParams?: Record< string, string | number >;
+};
+
+export function useFlowLoginUrl( {
+	flow,
+	loginUrlParams = {},
+	pageTitle,
+	returnStepSlug,
+	returnUrlParams = {},
+}: UseFlowLoginUrlProps ): string {
+	const locale = useFlowLocale();
+
+	const flowPath = flow.variantSlug ?? flow.name;
+	const isNonEnglishLocale = locale && locale !== 'en';
+	const returnQueryParams = {
+		...returnUrlParams,
+		...( isNonEnglishLocale ? { locale } : {} ),
+	};
+
+	const redirectTo = addQueryArgs(
```

Comment:
Since we have different logic "redirectTo" logics we can leave the logic outside the module.

My impression nowadays half the logic is inside the hook and half is handled externally.

Reply:
I disagree -- right now, the redirect logic is very inconsistent, especially regarding the URL parameters that are (not) included, so pushing this back onto developers feels like it will continue to cause headaches and functional gaps.

That said, it may make more sense to split that out into a separate hook so we can still make the code more consistent, but continue to separate concerns.

A lot of the complications here stem from the fact that I am trying to centralise code that is already inconsistent, and I am not 100% clear on what aspects should (and should not) be in the shared code.